src/index.ts

```diff
@@ -1,10 +1,16 @@
+import { createIoCContainer } from './ioc/index';
 import { Users } from './services/users';
 import { Logger } from './services/logger';
 
 import type { User, ApiConfig } from './types';
 
+const ioc = createIoCContainer();
+const usersService = ioc.resolve('users');
```

Comment:
you don't need to resolve the `usersService` here, because you resolve it below in `renderUsers` function.